storage/storage.go

```diff
@@ -1555,13 +1594,17 @@ var attrToFieldMap = map[string]string{
 // https://cloud.google.com/storage/docs/json_api/v1/how-tos/performance
 func (q *Query) SetAttrSelection(attrs []string) error {
```

Comment:
I think it might be cleaner to push all the gRPC logic down into toFieldMask, and push the JSON logic down into a new toFieldSelection() func. Then this func can just save the attrs passed in as `q.attrSelection []string` and we don't end up doing any wrong-transport work. What do you think?

Reply:
I agree, it felt dirty doing both in the same spot. Good plan, I'll do that.

Reply:
Ok. Done!